src/utilities/is.ts

```diff
@@ -8,7 +8,7 @@ function isObject(arg: unknown): arg is Record<string, unknown> {
 }
 
 function isFunction(arg: unknown): arg is Function {
-  return typeOf(arg) === 'Function';
+  return typeOf(arg) === 'Function' || typeOf(arg) === 'AsyncFunction';
```

Comment:
unfortunately lodash depends on `typeof` too in this case https://github.com/lodash/lodash/blob/master/isFunction.js. the custom method we use in https://github.com/ably-labs/spaces/blob/main/src/utilities/is.ts#L1-L3, which I've learned of only a couple of months ago, seems to quite more reliable, though I haven't done any extensive cross-browser testing.

Reply:
@dpiatek I believe lodash build process is not that simple :) 
<img width="1208" alt="image" src="https://github.com/ably-labs/spaces/assets/16488867/4c80e216-48a1-4ed1-927f-072122c30ddb">

Also it checks for `GeneratorFunction` and `Proxy` not only `Function` and `AsyncFunction`

Reply:
I'm going to assume we're sticking with what we have for now then?

Reply:
Maybe at least additionally add checks for `GeneratorFunction` and `Proxy`? What do you think, @dpiatek ?

Reply:
my bad @ttypic, I don't know why I thought it would be that simplistic haha

I'd opt to not introduce a new dependency at this time though. It's a lot more robust, but I don't think we need it at this time, and our implementation is a very basic version of `typeof` that lodash does anyway.

Reply:
@lmars as suggested by @ttypic I think it would be good to add `GeneratorFunction` and `Proxy` to the check

Reply:
ok np, done in 3d7a01c.